### DISCUSSION TITLE
How to handle inheritance using oneOf?

### DISCUSSION THREAD

**Discussion by davidjohnstonverint** [openapi-processor/openapi-processor-spring#138] [Q&A]
We have a problem we are trying to solve by using the `oneOf` attribute in OpenAPI in order to allow for a flexible portion of the request body. My expectation when using this was that the generated code would have a base class then sub-classes for each of the oneOf references, but I can't seem to get this to work. Is this scenario supported by the library or is there another way of achieving the same result?

An example of the OpenAPI spec is below:
```
    Foo:
      type: object
      properties:
        myProperties:
          $ref: '#/components/schemas/GenericProperties'
    GenericProperties:
      oneOf:
        - $ref: '#/components/schemas/SpecificPropertiesOne'
        - $ref: '#/components/schemas/SpecificPropertiesTwo'
    SpecificPropertiesOne:
      type: object
      properties:
        foo:
          type: string
          maxLength: 200
    SpecificPropertiesTwo:
      type: object
      properties:
        bar:
          type: string
          maxLength: 100
```

My expectation would be the following classes:

```
public class Foo {
  private GenericProperties myProperties;
}
public interface GenericProperties {}
public class SpecificPropertiesOne implements GenericProperties {
  private String foo;
  ...
}
public class SpecificPropertiesTwo implements GenericProperties {
  private String bar;
  ...
}
```

What I actually get is SpecificPropertiesOne and SpecificPropertiesTwo are generated but not referenced anywhere, and GenericProperties is not generated at all. And anywhere that references GenericProperties (e.g. myProperties in the Foo class) is actually just given the type Object.

**Comment by hauner** [comment]
that is currently not supported. :-(

I have an issue that goes into the same direction: https://github.com/openapi-processor/openapi-processor-core/issues/78 but I didn't need it often enough to start implementing it and so far nobody asked for it. ;-)

I will check how easy it is to implement.

What you could do is to use a type mapping for `GenericProperties` and provide the interface and implementation of `SpecificPropertiesOne` & `SpecificPropertiesTwo` yourself.  Obvious drawback is that you have to maintain it manually.
